### ISSUE TITLE
bottlejs typing for TypeScript

### ISSUE THREAD

**Issue by kenvunz** [young-steveo/bottlejs#46]
https://gist.github.com/kenvunz/5b0726d720ab5dff285b

Let me know what you think. Happy to do a PR for your repo and/or with your permission, I'll submit it to the [typings registry](https://github.com/typings/registry)

**Comment by young-steveo** [comment]
Interesting.  I'm not overly familiar with TypeScript, but I took a look at the documentation for writing .d.ts files, and I have a couple of questions:

1. [This line](https://gist.github.com/kenvunz/5b0726d720ab5dff285b#file-bottlejs-d-ts-L3) shows that the `Bottle.pop` function returns `void`, but it actually returns an instance of bottle.  Is this correct?
2. a Factory is passed a container object.  [This line](https://gist.github.com/kenvunz/5b0726d720ab5dff285b#file-bottlejs-d-ts-L33) shows the typing as `Object`, but I believe you could use the `IContainer` type that you defined.  Is that doable?

**Comment by kenvunz** [comment]
you are spot on! gist is updated!

```
static pop: (name?: string) => Bottle

...

factory(name: string, Factory: (container: Bottle.IContainer) => any): this;
```

**Comment by young-steveo** [comment]
As far as my limited TypeScript knowledge goes, everything looks good.

**Comment by zdhuka** [comment]
Is the type definition completed? I need it.

**Comment by kenvunz** [comment]
@young-steveo awesome!

to move forward, there are couple of options, and i'd leave it to you to decide

1. I'll make a PR that will add a `typings` property to `package.json` as outlined [here](https://github.com/Microsoft/TypeScript/wiki/Typings-for-npm-packages)

2. Submit to [typing registry](https://github.com/typings/registry)

What do you prefer? We could do both!

**Comment by young-steveo** [comment]
@kenvunz I'm okay with both. :+1:

**Comment by kenvunz** [comment]
- [x] Make a PR - see #47 

- [ ] Submit to [typing registry](https://github.com/typings/registry)

**Comment by young-steveo** [comment]
Now available in [v1.2.3](https://github.com/young-steveo/bottlejs/releases/tag/v1.2.3)
